[Failing Test]: JmsIOIT.testPublishingThenReadingAll highly flaky

### What happened?

Affecting https://ci-beam.apache.org/view/PostCommit/job/beam_PreCommit_Java_Jms_IO_Direct_Cron/

```
java.lang.AssertionError: actual number of records 14220 smaller than expected: 100000.
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.assertTrue(Assert.java:42)
	at org.apache.beam.sdk.io.jms.JmsIOIT.testPublishingThenReadingAll(JmsIOIT.java:195)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.apache.beam.sdk.testing.TestPipeline$1.evaluate(TestPipeline.java:323)
```

We should probably decrease the number of element when test is running locally

### Issue Failure

Failure: Test is flaky

### Issue Priority

Priority: 2 (backlog / disabled test but we think the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [ ] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

--- a/sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOIT.java
+++ b/sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOIT.java
@@ -26,8 +26,8 @@ import static org.junit.Assert.assertTrue;
 import java.io.IOException;
 import java.io.Serializable;
 import java.time.Instant;
-import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
 import java.util.UUID;
@@ -132,13 +132,15 @@ public class JmsIOIT implements Serializable {
 
   @Parameterized.Parameters(name = "with client class {3}")
   public static Collection<Object[]> connectionFactories() {
-    return Arrays.asList(
+    return Collections.singletonList(
         new Object[] {
           "vm://localhost", 5672, "jms.sendAcksAsync=false", ActiveMQConnectionFactory.class
-        },
-        new Object[] {
-          "amqp://localhost", 5672, "jms.forceAsyncAcks=false", JmsConnectionFactory.class
         });
+    // TODO(https://github.com/apache/beam/issues/26175) Test failure on direct runner due to
+    //  JmsIO read on amqp slow on Jenkins.
+    // new Object[] {
+    //   "amqp://localhost", 5672, "jms.forceAsyncAcks=false", JmsConnectionFactory.class
+    // });
   }
 
   private final CommonJms commonJms;
@@ -164,6 +166,8 @@ public class JmsIOIT implements Serializable {
       this.commonJms.startBroker();
       connectionFactory = this.commonJms.getConnectionFactory();
       connectionFactoryClass = this.commonJms.getConnectionFactoryClass();
+      // use a small number of record for local integration test
+      OPTIONS.setNumberOfRecords(10000);
     }
   }
 

